ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/export/InMemorySubmissionAggregator.scala

```diff
@@ -12,23 +14,26 @@ final class InMemorySubmissionAggregator(submissionInfo: SubmissionInfo, writer:
 
   import InMemorySubmissionAggregator._
 
-  private val buffer = mutable.ListBuffer.empty[WriteItem]
+  private val aggregate = mutable.SortedMap.empty[Key, Value]
 
-  override def addChild(): WriteSetBuilder = new InMemoryWriteSetBuilder(buffer)
+  override def addChild(): WriteSetBuilder = new InMemoryWriteSetBuilder(aggregate)
 
-  override def finish(): Unit = writer.write(submissionInfo, buffer)
+  override def finish(): Unit =
+    writer.write(submissionInfo, aggregate)
 }
 
 object InMemorySubmissionAggregator {
 
-  final class InMemoryWriteSetBuilder(buffer: mutable.Buffer[WriteItem]) extends WriteSetBuilder {
-    override def +=(data: WriteItem): Unit = buffer.synchronized {
-      buffer += data
+  final class InMemoryWriteSetBuilder private[InMemorySubmissionAggregator] (
+      aggregate: mutable.SortedMap[Key, Value],
```

Comment:
Why do you need to sort?

Reply:
This is done in [`820b782` (#7265)](https://github.com/digital-asset/daml/pull/7265/commits/820b7821282c949e2bc4e303d0bb9c73a771d770), which moves this logic to aggregation instead of doing it at the end.

I'm not sure why we sort the write set on export, but we always have.

Reply:
Hmm, I see. I guess this might or not be a problem in replay tests for non-determinism, depending on how they work. @miklos-da What do you think?

Reply:
I agree with this logic and would rather not sort them. If I can get the go-ahead to remove this logic, I'm happy to. I'd rather do it in this PR so we can get it released. However, I'm not comfortable making this decision solo; I'd like to get a thumbs-up from both you and @miklos-da.

Reply:
FYI, I tried removing sorting on both the write side and the integrity checker, and I get a bunch of ordering errors. So it's (unfortunately) necessary, at least for now.

I think removing it is out of scope for this PR.